AK/IPv6Address.h

```diff
@@ -111,7 +111,14 @@ class [[gnu::packed]] IPv6Address {
         if (string.is_null())
             return {};
 
-        auto const parts = string.split_view(':', SplitBehavior::KeepEmpty);
+        // NOTE: This supports URI syntax (square brackets) for IPv6 addresses.
+        //       See: https://datatracker.ietf.org/doc/html/rfc3986#section-3.2.2
+        if ((string.starts_with('[') && !string.ends_with(']')) || (string.ends_with(']') && !string.starts_with('[')))
+            return {};
+
+        auto const value_to_parse = string.replace("["sv, ""sv, ReplaceMode::FirstOnly).replace("]"sv, ""sv, ReplaceMode::FirstOnly);
```

Comment:
Bit of needless allocations here, we can just substring the input string. Something like:
```c++
auto starts_with_bracket = string.starts_with('[');
auto ends_with_bracket = string.ends_with(']');

if (starts_with_bracket || ends_with_bracket) {
    if (!starts_with_bracket || !ends_with_bracket)
        return {};

    string = string.substring_view(1, string.length() - 2);
}
```